sdk/identity/azure-identity/azure/identity/_internal/aadclient_certificate.py

```diff
@@ -19,14 +20,18 @@
 class AadClientCertificate(object):
     """Wraps 'cryptography' to provide the crypto operations AadClient requires for certificate authentication.
 
-    :param bytes pem_bytes: bytes of a a PEM-encoded certificate including the private key
+    :param bytes pem_bytes: bytes of a a PEM-encoded certificate including the (RSA) private key
     :param bytes password: (optional) the certificate's password
     """
     def __init__(self, pem_bytes, password=None):
         # type: (bytes, Optional[bytes]) -> None
+        private_key = serialization.load_pem_private_key(pem_bytes, password=password, backend=default_backend())
+        if not isinstance(private_key, RSAPrivateKey):
+            raise ValueError("CertificateCredential requires an RSA private key because it uses RS256 for signing")
+        self._private_key = cast(RSAPrivateKey, private_key)
```

Comment:
I'm assuming this cast is to appease mypy?

Reply:
I don't actually need the cast, it's a vestige of my first attempt, which mypy couldn't figure out:
```py
self._private_key = serialization.load_pem_private_key(pem_bytes, password=password, backend=default_backend())
if not isinstance(self._private_key, RSAPrivateKey):
    raise ValueError("CertificateCredential requires an RSA private key because it uses RS256 for signing")
```
I suppose because the assignment came before the `isinstance` check?